cmd/genji/insert.go

```diff
@@ -0,0 +1,140 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/dgraph-io/badger/v2"
+	"github.com/genjidb/genji"
+	"github.com/genjidb/genji/document"
+	"github.com/genjidb/genji/engine"
+	"github.com/genjidb/genji/engine/badgerengine"
+	"github.com/genjidb/genji/engine/boltengine"
+)
+
+func skipSpaces(r *bufio.Reader) (byte, error) {
+	var c byte
+	var err error
+
+	for {
+		c, err = r.ReadByte()
+		if err != nil {
+			return c, err
+		}
+
+		if c != '\n' && c != '\r' && c != ' ' && c != '\t' {
+			break
+		}
+	}
+
+	return c, nil
+}
+
+func executeInsertCommand(db *genji.DB, table string, r io.Reader) error {
+	q := fmt.Sprintf("INSERT INTO %s VALUES ?", table)
+	rd := bufio.NewReader(r)
+	var c byte
+	var err error
+
+	// Ignore spaces.
+	c, err = skipSpaces(rd)
+	if err != nil {
+		return err
+	}
+	switch c {
+	case '{': // Json objects
+		if err := rd.UnreadByte(); err != nil {
+			return err
+		}
+
+		dec := json.NewDecoder(rd)
+		// NewFromDecoder is a wrapper to call parseJsonDocument.
+		d, err := document.NewFromDecoder(dec)
```

Comment:
This is not necessary, document.FieldBuffer has an [UnmarshalJSON ](https://pkg.go.dev/github.com/genjidb/genji@v0.7.1/document#FieldBuffer.UnmarshalJSON)method, you can simply decode to it

```go
var fb document.FieldBuffer

err := dec.Decode(&fb)

db.Exec(&fb)
```

Reply:
I love this solution 👍